### PR TITLE
fix query plan for wrapped types

### PR DESCRIPTION
**Pull request by crissi**
Was getting error: "Call to undefined method GraphQL\Type\Definition\ListOfType::getField()" when using QueryPlan::lookAhead()

### CODE REVIEW

File: tests/Type/QueryPlanTest.php [hunk 1]
```diff
@@ -298,6 +298,78 @@ public function testQueryPlan() : void
...
+        $expectedQueryPlan = [
+            'id' => [
+                'type' => Type::string(),
+                'args' => [],
+                'fields' => [],
+            ],
+        ];
+
+        $hasCalled = false;
+        /** @var QueryPlan $queryPlan */
```
Comment (spawnia) [inline tests/Type/QueryPlanTest.php:326]:
```suggestion
        /** @var QueryPlan|null $queryPlan */
```

File: tests/Type/QueryPlanTest.php [hunk 2]
```diff
@@ -298,6 +298,78 @@ public function testQueryPlan() : void
...
+                        ];
+                    },
+                ],
+            ],
+        ]);
+
+        $schema = new Schema(['query' => $blogQuery]);
+        $result = GraphQL::executeQuery($schema, $doc)->toArray();
+
+        self::assertTrue($hasCalled);
```
Comment (spawnia) [inline tests/Type/QueryPlanTest.php:358]: I don't think this is necessary.

File: tests/Type/QueryPlanTest.php [hunk 3]
```diff
@@ -298,6 +298,78 @@ public function testQueryPlan() : void
...
+        self::assertEquals(
```
Comment (spawnia) [inline tests/Type/QueryPlanTest.php:359]: That is also superfluous. How about you just return `null` from the resolver and don't even check the result? That makes the test more focused.

**Review by spawnia — CHANGES_REQUESTED** [review]

File: tests/Type/QueryPlanTest.php [hunk 4]
```diff
@@ -298,6 +298,60 @@ public function testQueryPlan() : void
...
+                        $queryPlan = $info->lookAhead();
+
+                        return [];
+                    },
+                ],
+            ],
+        ]);
+
+        $schema = new Schema(['query' => $blogQuery]);
+        $result = GraphQL::executeQuery($schema, $doc)->toArray();
```
Comment (spawnia) [inline tests/Type/QueryPlanTest.php:350]:
```suggestion
        GraphQL::executeQuery($schema, $doc);
```

**Review by spawnia — APPROVED** [review]

File: tests/Type/QueryPlanTest.php [hunk 5]
```diff
@@ -298,6 +298,60 @@ public function testQueryPlan() : void
...
+                        return [];
+                    },
+                ],
+            ],
+        ]);
+
+        $schema = new Schema(['query' => $blogQuery]);
+        GraphQL::executeQuery($schema, $doc);
+
+        self::assertEquals($expectedQueryPlan, $queryPlan->queryPlan());
```
Comment (simPod) [inline tests/Type/QueryPlanTest.php:352]: well mb we should use `assertSame` or another set of assertions. `assertEquals` might incorrectly compare objects which might lead to bugs when test fixtures are modified in the future.

**Review by simPod — APPROVED** [review]

File: tests/Type/QueryPlanTest.php (hunk 5 above)
Comment (crissi) [reply tests/Type/QueryPlanTest.php:352]: okay will change

**Comment by spawnia** [comment]
The implementation of is fine, but i would like @vladar to confirm that this is indeed how the query plan should behave for wrapped types.

**Review by vladar — APPROVED** [review]
Agree, this is a valid bugfix! Thanks! 👍

**Pull request merged**